src/shaders/_prelude.fragment.glsl

```diff
@@ -29,3 +29,14 @@ vec3 srgb_to_linear(vec3 color) {
 vec3 gamma_mix(vec3 a, vec3 b, float x) {
     return linear_to_srgb(mix(srgb_to_linear(a), srgb_to_linear(b), x));
 }
+
+highp vec3 hash(highp vec2 p) {
+    highp vec3 p3 = fract(p.xyx * vec3(443.8975, 397.2973, 491.1871));
+    p3 += dot(p3, p3.yxz + 19.19);
+    return fract((p3.xxy + p3.yzz) * p3.zyx);
+}
+
+vec3 dither(vec3 color, highp vec2 seed) {
```

Comment:
Is there any risk of applying this function twice for a given fragment position in our codebase, and would this be a problem if we do?

Usually, such a function is applied in a very late stage/post process ensuring unique processing, but since we're applying it a bit differently in our code I wonder if it's possible for a single screen position to have it applied multiple times and if it resulted in some issues.

Reply:
That's a good point, but I think the rule was to apply it at the last possible moment _before quantization_. Since the framebuffers (correct me if I'm wrong) are 8-bit channels, am I correct in understanding that since we apply this per-pass, this is already the last possible moment at which we can apply it?

Reply:
Yes that makes sense. I was thinking of the case where we would read back from a previous drawn pass to blend with a previous result. But as long as we do it before the quantized step that seems correct!

Reply:
Are there cases where we read back in data which has been rendered to texture, apart from tiles which exclude fog anyway?